src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -328,5 +329,60 @@ public static int GetTrailingWhitespaceLength(this SyntaxNode node, FormattingCo
     }
 
     public static SyntaxNode? FindInnermostNode(this SyntaxNode node, int index, bool includeWhitespace = false)
-        => node.FindToken(index, includeWhitespace)?.Parent;
+    {
+        var token = node.FindToken(index, includeWhitespace);
+
+        // If the index is EOF but the node has index-1,
+        // then try to get a token to the left of the index.
+        // patterns like
+        // <button></button>$$
+        // should get the button node instead of the razor document (which is the parent
+        // of the EOF token)
+        if (token.Kind == SyntaxKind.EndOfFile && node.Span.Contains(index - 1))
```

Comment:
I'm not sure this is the _right_ answer. Would love thoughts on this.

Reply:
i guess this makes sense. if there were whitespace at the end, i think the compiler method would do the same thing. Maybe @333fred has opinions though